cclib/method/ddec.py

```diff
@@ -88,17 +96,36 @@ def _read_proatom(self, directory, atom_num, charge):
         #       atoms.h5
         # File format:
         #   Starting from line 13, each line contains the charge densities for each shell
-        chargemol_path = os.path.join(
+        # If `charge` is not an integer, proatom densities have to be linearly interpolated between
+        # the densities of the ion/atom with floor(charge) and ceiling(charge)
+        charge_floor = int(math.floor(charge))
+        charge_ceil = int(math.ceil(charge))
+        
+        chargemol_path_floor = os.path.join(
+            directory,
+            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge_floor),
+        )
+        chargemol_path_ceil = os.path.join(
             directory,
-            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge),
+            "c2_{:03d}_{:03d}_{:03d}_500_100.txt".format(atom_num, atom_num, atom_num - charge_ceil),
         )
         horton_path = os.path.join(directory, "atoms.h5")
 
-        if os.path.isfile(chargemol_path):
+
+        if os.path.isfile(chargemol_path_floor) or os.path.isfile(chargemol_path_ceil):
             # Use chargemol proatom densities
             # Each shell is .05 angstroms apart (uniform).
             # *scalefactor* = 10.58354497764173 bohrs in module_global_parameter.f08
-            density = numpy.loadtxt(chargemol_path, skiprows=12, dtype=float)
+            if atom_num <= charge_floor:
+                density_floor = numpy.array([0])
```

Comment:
Coverage says this block isn't being tested. I still don't think it's right, but don't know what it should be.

Reply:
This block wouldn't be tested in the test suite because `chargemol` densities cannot be included in the code as discussed in   #914. That could be why the block is being tested?

Reply:
Yes, that's why. Just pointing out that this is dead code (for now at least).